### PR TITLE
include custom event polyfill in core

### PR DESCRIPTION
**Pull request by teleaziz**

### CODE REVIEW

**Comment by teleaziz** [comment]
closing for testing, will reopen later

File: packages/core/src/builder.class.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import 'custom-event-polyfill';
```
Comment (steve8708) [inline packages/core/src/builder.class.ts:1]: I actually vaguely recall us using this then removing it due to a problem. can you check the source code to see if this is identical to what we used in the webcomponents? or possibly check the git history as to when we changed this over to be manually copy + pasted and edited

my vague recollection was it defining the CustomEvent even if it already was defined or something. or not being compatible server side (not checking for window existing) first

**Comment by teleaziz** [comment]
This lib is similar to the polyfill we have in,  except it handles preventing default on IE:
https://github.com/kumarharsh/custom-event-polyfill/blob/813029a466f0a0c7174db86656634ea2c74b1011/polyfill.js#L16-L18

checking history now

File: packages/core/src/builder.class.ts (hunk 1 above)
Comment (teleaziz) [reply packages/core/src/builder.class.ts:1]: ok in this case will use the old polyfill

File: packages/core/src/polyfills/custom-event-polyfill.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+;(function() {
```
Comment (steve8708) [inline packages/core/src/polyfills/custom-event-polyfill.js:1]: we will need to check that window exists too - webcomponents only loads in browser but core loads everywhere (browser, server, native, possibly even webworker)

so just a `if (typeof window === 'undefined') { return }` should suffice

Comment (teleaziz) [reply packages/core/src/polyfills/custom-event-polyfill.js:1]: it's in the last commit, but will update to use this syntax

Comment (steve8708) [reply packages/core/src/polyfills/custom-event-polyfill.js:1]: oh oops I missed that (clicked via email), what you have totally works

**Pull request merged**